judo-ui-react/src/main/resources/actor/src/components/table/LazyTable.tsx.hbs

```diff
@@ -0,0 +1,814 @@
+      { Object.keys(additionalToolbarActions).map((key) => {
+        const AdditionalButton = additionalToolbarActions[key];
+        return <AdditionalButton key={key} />;
+      }) }
+    </>;
+  };
+
+  async function filterAction(id: string) {
+    const filterResults = await actions[calculateActionName(relationName, 'filterAction')]!(id, filterOptions, filterModel, filters);
+  {{# if isUseInlineColumnFilters }}
+    if (filterResults.model) {
+      handleFilterModelChange({
+        ...filterResults.model!,
+      });
+    }
+  {{ else }}
+    if (Array.isArray(filterResults.filters)) {
+      handleFiltersChange([...filterResults.filters!]);
+    }
+  {{/ if }}
+  }
+
+  const getTableButtonVisibility = useCallback((expression: string, actionName: string): boolean => {
+    if (expression === undefined) {
+      return false;
+    }
+
+    if (expression === 'false') {
+      return false;
+    }
+
+    if (expression === 'isFormUpdateable()') {
+      return isFormUpdateable ? isFormUpdateable() : false;
+    }
+
+    let result: boolean = data.length !== undefined && data.length > 0;
+    if (expression.includes(`ownerData.${enabledByName}`)) {
+      result = result && ownerData && ownerData['enabledByName'];
+    }
+
+    if (expression.includes('isFormUpdateable()')) {
+      return result && (isFormUpdateable ? isFormUpdateable(): false);
+    }
+
+    if (actionName.includes('clearAction') || actionName.includes('ClearAction')) {
+      return result;
+    }
+
+    if (expression === 'selectionModel.length > 0') {
+      return selectionModel.length > 0;
+    }
+
+    return true;
+  }, [selectionModel, ownerData, isFormUpdateable, data]);
+
+  const getTableRowButtonDisabledConditions = useCallback((row: T): boolean => {
+    let expression = data.find((r) => r.__identifier === row.__identifier )?.__identifier;
+    if (expression === undefined) {
+      return false;
+    }
+
+    if (expression === 'isLoading') {
+      return isLoading;
+    }
+
+    if (expression === 'getSelectedRows().length > 0 || isLoading') {
+      return getSelectedRows().length > 0 || isLoading;
+    }
+
+    let result: boolean = getSelectedRows().length > 0;
+    if (expression.includes('isFormUpdateable() || ')) {
+      result = result || (isFormUpdateable ? isFormUpdateable() : false);
+    }
+
+    if (expression === 'editMode || ') {
+      result = result || (editMode ? editMode : false);
+    }
+
+    if (expression.includes(`!ownerData.${enabledByName} || `)) {
+      result = result || (ownerData ? !ownerData['enabledByName'] : false);
+    }
+
+    if (expression.includes('!row.__deleteable || ')) {
+      result = result || !row.__deleteable;
+    }
+
+    if (expression.includes(`!row.${enabledByName} || `)) {
+      return data.length || row['enabledByName'];
+    }
+
+    return result || isLoading;
+  }, [data, editMode, ownerData, isLoading]);
+
+  function calculateActionName(relationName: string, actionName: string) {
```

Comment:
This should be extracted.

Reply:
[f51c46b](https://github.com/BlackBeltTechnology/judo-ui-react-template/pull/365/commits/f51c46bc6ce006871f020c8d52167c195794f251)